src/main/java/io/javalin/core/JavalinServlet.kt

```diff
@@ -74,7 +77,25 @@ class JavalinServlet(
                 }
             }
             if (endpointEntries.isEmpty() && type != HandlerType.HEAD && type != HandlerType.GET) {
-                throw HaltException(404, "Not found")
+
+                if (prefer405over404) {
+                    val availableHandlerTypes = MethodNotAllowedUtil.findAvailableHandlerTypes(matcher, requestUri)
+
+                    if (availableHandlerTypes.isEmpty()) {
+                        throw HaltException(404, "Not found")
+                    }
+
+                    val acceptableReturnTypes = ctx.header("Accept")
+
+                    val body = if (acceptableReturnTypes != null && acceptableReturnTypes.contains("application/json")) {
```

Comment:
What if acceptable return type is `application/yaml`?
Maybe it is a better idea to just send a `HaltException` with simple text by default?
Then user can override it in exception mapper or so.

Reply:
I think making special cases for HTML and JSON is okay, they will account for something like 99.9% of all requests. A general string formatted as JSON would also work for everything though.